uat/custom-components/src/main/java/com/aws/greengrass/artifacts/ShadowComponent.java

```diff
@@ -0,0 +1,221 @@
+                    switch (operation) {
+                        case "GetThingShadow":
+                            handleGetThingShadowOperation(thingName, shadowName);
+                            break;
+                        case "UpdateThingShadow":
+                            handleUpdateThingShadowOperation(thingName, shadowName, shadowDocumentPayload);
+                            break;
+                        case "DeleteThingShadow":
+                            handleDeleteThingShadowOperation(thingName, shadowName);
+                            break;
+                        case "SetupListNamedShadowTest":
+                            handleSetupForListNamedShadows(thingName, shadowDocumentPayload);
+                            break;
+                        case "ListNamedShadowsForThing":
+                            handleListNamedShadowsForThingOperation(thingName, pageSize, nextToken);
+                            break;
+                        case "NoOp":
+                            // Do nothing
+                            break;
+                        default:
+                            LOGGER.error("No matching operation found for: {}", operation);
+                            throw new ExecutionException(new UnsupportedOperationException("No matching operation found for: " + operation));
+                    }
+                    isSuccessful = true;
+                } catch (ExecutionException e) {
+                    if (e.getCause() instanceof InvalidArgumentsError) {
+                        String errorMessage = String.format("InvalidArgumentsError error in the %s operation", operation);
+                        LOGGER.error(errorMessage, e);
+                    } else if (e.getCause() instanceof ResourceNotFoundError) {
+                        String errorMessage = String.format("ResourceNotFoundError error in the %s operation", operation);
+                        LOGGER.error(errorMessage, e);
+                    } else if (e.getCause() instanceof ServiceError) {
+                        String errorMessage = String.format("ServiceError error in the %s operation", operation);
+                        LOGGER.error(errorMessage, e);
+                    } else if (e.getCause() instanceof UnauthorizedError) {
+                        String errorMessage = String.format("UnauthorizedError error in the %s operation", operation);
+                        UnauthorizedError unauthorizedError = (UnauthorizedError) e.getCause();
+                        LOGGER.error(errorMessage, e);
+                    } else {
+                        String errorMessage = String.format("Execution error in the %s operation", operation);
+                        LOGGER.error(errorMessage, e);
+                    }
+                    TimeUnit.SECONDS.sleep(1);
+                }
+            } while (!isSuccessful && finalTime.isAfter(Instant.now()));
+        } catch (InterruptedException | IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)